CMakeLists.txt

```diff
@@ -2,6 +2,12 @@ cmake_minimum_required(VERSION 3.13)
 
 project(oxenmq-node)
 
+include(CheckCXXSourceCompiles)
+include(CheckLibraryExists)
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
A preferable modern CMake way to do this is to set the CXX_* properties on the target rather than as globals, like this: https://github.com/oxen-io/oxen-mq/blob/ff0e515c5187b1141ac9acb0a5a100a99ca1bee7/CMakeLists.txt#L142

Reply:
this is cmake.js, not modern cmake keep in mind.

Reply:
will add